libs/postgres_connection/src/lib.rs

```diff
@@ -0,0 +1,222 @@
+
+    /// Return a `<host>:<port>` string.
+    pub fn raw_address(&self) -> String {
+        format!("{}:{}", self.host(), self.port())
+    }
+
+    /// Build a client library-specific connection configuration.
+    /// Used for testing and when we need to add some obscure configuration
+    /// elements at the last moment.
+    pub fn to_tokio_postgres_config(&self) -> tokio_postgres::Config {
+        // Use `tokio_postgres::Config` instead of `postgres::Config` because
+        // the former supports more options to fiddle with later.
+        let mut config = tokio_postgres::Config::new();
+        config.host(&self.host().to_string()).port(self.port);
+        if let Some(password) = &self.password {
+            config.password(password);
+        }
+        if !self.options.is_empty() {
+            // These options are command-line options and should be escaped before being passed
+            // as an 'options' connection string parameter, see
+            // https://www.postgresql.org/docs/15/libpq-connect.html#LIBPQ-CONNECT-OPTIONS
+            //
+            // They will be space-separated, so each space inside an option should be escaped,
+            // and all backslashes should be escaped before that. Although we don't expect options
+            // with spaces at the moment, they're supported by PostgreSQL. Hence we support them
+            // in this typesafe interface.
+            config.options(
+                &self
+                    .options
+                    .iter()
+                    .map(|s: &String| s.replace('\\', "\\\\").replace(' ', "\\ "))
+                    .collect::<Vec<_>>()
+                    .join(" "),
+            );
+        }
+        config
+    }
+
+    /// Connect using postgres protocol with TLS disabled.
+    pub fn connect_no_tls(&self) -> Result<postgres::Client, postgres::Error> {
+        postgres::Config::from(self.to_tokio_postgres_config()).connect(postgres::NoTls)
+    }
+}
+
+impl fmt::Debug for PgConnectionConfig {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        // We want `password: Some(REDACTED-STRING)`, not `password: Some("REDACTED-STRING")`
+        // so even if the password is `REDACTED-STRING` (quite unlikely) there is no confusion.
+        // Hence the custom formatter.
+        struct Redacted {}
+        impl fmt::Debug for Redacted {
+            fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+                f.write_str("REDACTED-STRING")
+            }
+        }
+
+        f.debug_struct("PgConnectionConfig")
+            .field("host", &self.host)
+            .field("port", &self.port)
+            .field("password", &self.password.as_ref().map(|_| Redacted {}))
+            .finish()
+    }
+}
+
+#[cfg(test)]
+mod tests_pg_connection_config {
+    use crate::PgConnectionConfig;
+    use once_cell::sync::Lazy;
+    use url::Host;
+
+    static STUB_HOST: Lazy<Host> = Lazy::new(|| Host::Domain("stub.host.example".to_owned()));
```

Comment:
Do you really need to have the dev-dependency for this line? 

maybe you can replace it with a short method in this module, can't you?

```rust
fn stub_host() -> Host  {
    Host::Domain("stub.host.example".to_owned())
}
```

Reply:
I have something different on my mind. [`Host`](https://docs.rs/url/latest/url/enum.Host.html) can be parameterized by `&str` instead of String, so can make `new_host_port` more generic and  call [`to_owned`](https://docs.rs/url/latest/url/enum.Host.html#method.to_owned) inside.

This will let us use

```rust
const STUB_HOST: Host<&str> = Host::Domain("stub.host.example");
```

API users will likely benefit from this too.

Reply:
More elegant proposal than mine! 

My idea was devoted not to adding new dev-dependency for this one line, but

Reply:
Nice idea! Unfortunately, I bumped into the comparison code: `Host<String>` can only be compared with `Host<String>`, not `Host<&str>`. Therefore I have to convert `STUB_HOST: Host<&str>` to `Host<String>` in `assert_eq!` anyway. Even if I support both `Host<String>` and `Host<&str>` in `new_host_port` (the only way to "overload" in Rust is to create a trait, isn't it..?), I still get `.to_owned()` in the test.

So I prefer to leave the code as is. What I have here is a global const, initialized once, not a value generated dynamically on the fly for each test (e.g. `assert_eq!` do not clone the const), so `Lazy` expresses the intention better. Also, we use `once_cell` heavily in our code a lot already, and its standard counterpart is in progress (https://github.com/rust-lang/rust/issues/74465).

Reply:
Ok! It's definitely a nice-to-have, not a must. I might revisit this later on my own.